### PR TITLE
Update jax & jaxlib versions

### PR DESCRIPTION
**Pull request by jonas-eschle**
Update to the newest versions (skipping some intermediate jaxlib versions that have effectively no effect
Deprecating very old version

I am not exactly sure on the CUDA dependencies, let's see.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Related to #42697

**Comment by jonas-eschle** [comment]
@spackbot fix style

**Comment by adamjstewart** [comment]
#42697 is merged now. If you still want to add more versions of JAX, you can rebase. Or if the latest version is all you need, you can close.

**Comment by jonas-eschle** [comment]
@adamjstewart thanks, I've updated it and added the newest version (and more fine-grained cuda requirements), i.e. ready from my side

File: var/spack/repos/builtin/packages/py-jaxlib/package.py [hunk 1]
```diff
@@ -62,11 +72,15 @@ class PyJaxlib(PythonPackage, CudaPackage):
     depends_on("bazel@4.1.0", when="@0.1.70:0.1.74", type="build")
 
     # README.md
-    depends_on("cuda@11.4:", when="@0.4:+cuda")
+    depends_on("cuda@11.8:", when="@0.4.11:+cuda")
+    depends_on("cuda@11.4:", when="@0.4.0:0.4.7+cuda")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jaxlib/package.py:82]: Where do these numbers come from? Want to document how to update this package in the future.

File: var/spack/repos/builtin/packages/py-jaxlib/package.py [hunk 2]
```diff
@@ -62,11 +72,15 @@ class PyJaxlib(PythonPackage, CudaPackage):
...
     depends_on("cuda@11.1:", when="@0.3+cuda")
     # https://github.com/google/jax/issues/12614
     depends_on("cuda@11.1:11.7.0", when="@0.1+cuda")
-    depends_on("cudnn@8.2:", when="@0.4:+cuda")
+
+    depends_on("cudnn@8.8:", when="@0.4.11:+cuda")
+    depends_on("cudnn@8.2:", when="@0.4:0.4.7+cuda")
+    depends_on("cudnn@8.2:", when="@0.4:0.4.7+cuda")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jaxlib/package.py]: Duplicated?

**Review by adamjstewart — COMMENTED** [review]
CI will likely fail on PowerPC because of https://github.com/google/jax/issues/19992. We need to figure out the most recent version of jaxlib that builds on pcc64le.

File: var/spack/repos/builtin/packages/py-jaxlib/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-jaxlib/package.py:82]: So this is just based on:
```python
"nvidia-cuda-runtime-cu11>=11.8",                                                
"nvidia-cudnn-cu11>=8.8",
```
? Works for me

**Review by adamjstewart — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-jaxlib/package.py (hunk 1 above)
Comment (jonas-eschle) [reply var/spack/repos/builtin/packages/py-jaxlib/package.py:82]: Yes, exactly, based on that (sorry, forgot to answer)

File: var/spack/repos/builtin/packages/py-jaxlib/package.py (hunk 2 above)
Comment (jonas-eschle) [reply var/spack/repos/builtin/packages/py-jaxlib/package.py]: yes, removed

**Comment by adamjstewart** [comment]
Just need to get the tests to pass now.

**Comment by alalazo** [comment]
There is a legit issue building on Power https://gitlab.spack.io/spack/spack/-/jobs/10186759

**Comment by adamjstewart** [comment]
Yep, see above:

> CI will likely fail on PowerPC because of google/jax#19992. We need to figure out the most recent version of jaxlib that builds on pcc64le.

**Comment by adamjstewart** [comment]
Somehow just noticed https://github.com/google/jax/commit/8be057de1f50756fe7522f7e98b2f30fad56f7e4, which enforces that the jax version must be >= the jaxlib version. This is going to be a PITA to document properly...

**Comment by adamjstewart** [comment]
I added a basic check for this in #43062 but it requires us to iterate over all versions in the package. So we just need to add every new version to this list every time. Wish we had a better way to handle this.

**Comment by jonas-eschle** [comment]
@adamjstewart can you re-approve? There were merge conflicts that I've just resolved

**Review by adamjstewart — APPROVED** [review]
Still likely need to expand the range of versions for which ppc64le conflicts in order to satisfy CI.

**Review by adamjstewart — DISMISSED** [review]
Still likely need to expand the range of versions for which ppc64le conflicts in order to satisfy CI.

**Review by adamjstewart — DISMISSED** [review]

**Review by adamjstewart — DISMISSED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**